src/mongocxx/collection.cpp

```diff
@@ -65,6 +65,23 @@ stdx::string_view collection::name() const noexcept {
     return _impl->name;
 }
 
+void collection::name(stdx::string_view new_name, bool overwrite_old) {
+  bson_error_t error;
+
+  _impl->name = (std::string)new_name;
```

Comment:
Few more comments here: `mongoc_collection_t` doesn't have a name field, I think you meant `->collection`. However, `mongoc_collection_rename` copies the new name to the impl collection field, so you won't need this. (see additional comment below)